src/main/java/org/killbill/billing/catalog/api/VersionedCatalog.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2014-2018 Groupon, Inc
+ * Copyright 2014-2018 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.catalog.api;
+
+import java.util.List;
+
+// Static catalog API methods must operate on the latest ("current") version
+public interface VersionedCatalog<CatalogType> extends StaticCatalog, Catalog {
```

Comment:
Doesn't this conflict with our [implementation class](https://github.com/killbill/killbill/blob/killbill-0.19.16/catalog/src/main/java/org/killbill/billing/catalog/VersionedCatalog.java#L74) ?

Reply:
I see you renamed `VersionedCatalog` -> `DefaultVersionedCatalog` inside KB.